homeassistant/components/cmus/media_player.py

```diff
@@ -65,33 +65,67 @@ def setup_platform(hass, config, add_entities, discover_info=None):
     name = config[CONF_NAME]
 
     try:
-        cmus_remote = CmusDevice(host, password, port, name)
+        cmus_remote = CmusRemote(server=host, port=port, password=password)
     except exceptions.InvalidPassword:
         _LOGGER.error("The provided password was rejected by cmus")
         return False
-    add_entities([cmus_remote], True)
+    add_entities([CmusDevice(device=cmus_remote, name=name, server=host)], True)
+
+
+class CmusRemote:
+    """Representation of a cmus connection."""
+
+    def __init__(self, server, port, password):
+        """Initialize the cmus remote."""
+
+        self._server = server
+        self._port = port
+        self._password = password
+        self.cmus = self.connect()
+
+    def connect(self):
+        """Connect to the cmus server."""
+
+        try:
+            cmus = remote.PyCmus(
+                server=self._server, port=self._port, password=self._password
+            )
+        except exceptions.InvalidPassword:
+            _LOGGER.error("The provided password was rejected by cmus")
+
+        return cmus
```

Comment:
This will error if the exception is triggered. Then `cmus` won't be defined.

Reply:
Thanks - good catch.